zebra-state/src/service/finalized_state/disk_format/upgrade/cache_genesis_roots.rs

```diff
@@ -0,0 +1,181 @@
+//! Updating the genesis note commitment trees to cache their roots.
+//!
+//! This reduces CPU usage when the genesis tree roots are used for transaction validation.
+//! Since mempool transactions are cheap to create, this is a potential remote denial of service.
+
+use std::sync::mpsc;
+
+use zebra_chain::{block::Height, sprout};
+
+use crate::service::finalized_state::{disk_db::DiskWriteBatch, ZebraDb};
+
+use super::CancelFormatChange;
+
+/// Runs disk format upgrade for changing the sprout and history tree key types.
+///
+/// Returns `Ok` if the upgrade completed, and `Err` if it was cancelled.
+///
+/// # Panics
+///
+/// If the state is empty.
+#[allow(clippy::unwrap_in_result)]
+#[instrument(skip(upgrade_db, cancel_receiver))]
+pub fn run(
+    initial_tip_height: Height,
```

Comment:
Why don't we need to prefix `initial_tip_height` with an underscore since we don't use it anywhere?

Reply:
Because it is used by the instrument macro. Happy to use an underscore of you think that would be clearer?

Reply:
I think it would because I noticed the function takes `initial_tip_height`, and then I noticed it uses `sprout_tree_for_tip()`, and I was thinking if `initial_tip_height` could be obsolete by the time `sprout_tree_for_tip()` is called. Then I noticed we don't use `initial_tip_height` anywhere. Having it prefixed would be more explicit about a situation like this.